Content.Server/_NF/SizeAttribute/SizeAttributeSystem.cs

```diff
@@ -0,0 +1,95 @@
+using System.Numerics;
+using Content.Server.SizeAttribute;
+using Robust.Server.GameObjects;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Collision.Shapes;
+using Robust.Shared.Physics.Systems;
+using Content.Shared.Item.PseudoItem;
+using Content.Shared.Nyanotrasen.Item.PseudoItem;
+
+namespace Content.Server._NF.SizeAttribute
+{
+    public sealed class SizeAttributeSystem : EntitySystem
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+        [Dependency] private readonly AppearanceSystem _appearance = default!;
+        [Dependency] private readonly FixtureSystem _fixtures = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SizeAttributeComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, SizeAttributeComponent component, ComponentInit args)
+        {
+            if (!TryComp<SizeAttributeWhitelistComponent>(uid, out var whitelist))
```

Comment:
Since by introducing this system to Metempsychosis means having the possibility for entities to be handed a SizeAttribute after character creation, you now need to account for SizeAttribute being handed to an entity that potentially doesn't have SizeAttributeWhitelist. Since we're exiting out if a species isn't whitelisted for SizeAttribute, you might as well cleanup by RemComping the component if for whatever reason it was handed to a species lacking the whitelist. That will also be useful if for example someone hits AddComponent in VV without first doing so with SizeAttributeWhitelist.